e2e/diff/no_schema_diff_test.go

```diff
@@ -0,0 +1,98 @@
+package e2e
+
+import (
+	"context"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"ariga.io/atlas/sql/migrate"
+	"entgo.io/ent/dialect/sql/schema"
+
+	entMigrate "github.com/openmeterio/openmeter/internal/ent/db/migrate"
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+var PG_URL string
+
+func TestMain(m *testing.M) {
+	PG_URL = os.Getenv("POSTGRES_URL")
+
+	os.Exit(m.Run())
+}
+
+const (
+	DIR = "./tmp/"
+)
+
+func setup(t *testing.T) {
+	if PG_URL == "" {
+		t.Skip("POSTGRES_URL not set")
+	}
+}
+
+func TestNoSchemaDiffOnMigrate(t *testing.T) {
+	setup(t)
+	driver, err := entutils.GetPGDriver(PG_URL)
+	if err != nil {
+		t.Fatalf("failed to get pg driver %s", err)
+	}
+
+	// OpenMemDir doesn't work from inside dagger
+	// dir := migrate.OpenMemDir("migrations")
+	err = os.RemoveAll(DIR)
+	if err != nil {
+		t.Fatalf("failed to remove dir %s", err)
+	}
+	err = os.MkdirAll(DIR, os.ModePerm)
+	if err != nil {
+		t.Fatalf("failed to create dir %s", err)
+	}
+	dir, err := migrate.NewLocalDir(DIR)
+	if err != nil {
+		t.Fatalf("failed to open local dir %s", err)
+	}
+
+	migrate, err := schema.NewMigrate(driver, schema.WithDir(dir))
+	if err != nil {
+		t.Fatalf("failed to create migrate %s", err)
+	}
+	err = migrate.Diff(context.Background(), entMigrate.Tables...)
+	if err != nil {
+		t.Fatalf("failed to diff schema %s", err)
+	}
+
+	files, err := dir.Files()
```

Comment:
I'm not sure if that's possible here, but if it is, using some dir walking method would be more elegant.

Reply:
We could do a `filepath.Walk(...)` directly on the os for the tmp dir, but after using `.MkdirTemp()` the contents will always be flat